pkg/streamingpromql/operators/binops/grouped_vector_vector_binary_operation.go

```diff
@@ -416,7 +416,9 @@ func (g *GroupedVectorVectorBinaryOperation) manySideGroupKeyFunc() func(manySid
 	labelsToRemove := g.VectorMatching.Include
 
 	if g.shouldRemoveMetricNameFromManySide() {
+		labelsToRemove = make([]string, 0, len(g.VectorMatching.Include)+1)
```

Comment:
Not for this change, but should we consider using a string pool?

Reply:
Would be interesting to try, but I suspect the impact will be small given we tend to only use string slices for things that are created once per query (eg. this list of label names).